src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -215,8 +215,25 @@ public void stop(@NonNull Throwable cause) throws Exception {
     @Override public void onResume() {
         try {
             if (state == null) {
-                Run<?, ?> run = getContext().get(Run.class);
-                LOGGER.fine(() -> "No ExecutorStepDynamicContext found for node block in " + run + "; perhaps loading from a historical build record, hoping for the best");
+                var flowNode = getContext().get(FlowNode.class);
+                LOGGER.fine(() -> "node block " + getContext() + " not yet scheduled, checking for an existing queue item");
+                if (flowNode == null) {
+                    LOGGER.fine(() -> "No FlowNode found for node block " + getContext() + "; can't recover" );
+                } else {
+                    var action = flowNode.getAction(QueueItemActionImpl.class);
```

Comment:
To remind myself: this would have been added right away in https://github.com/jenkinsci/workflow-durable-task-step-plugin/blob/6ae8123a901592f1c4ce1920d4968826a2d8d665/src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java#L132-L139